tools/fpgaconf/fpgaconf.c

```diff
@@ -389,10 +390,21 @@ int read_bitstream(char *filename, struct bitstream_info *info)
 	FILE *f;
 	long len;
 	int ret;
+	struct stat file_mode = { 0 };;
 
 	if (!filename || !info)
 		return -EINVAL;
 
+	if (stat(filename, &file_mode) < 0) {
+		perror(filename);
+		return -1;
+	}
+
```

Comment:
Klocwork may also call out that the attributes / existence of the file referred to by `filename` may change between the `stat()` call and a subsequent `open()` (similar, though not exactly identical to [this](https://wiki.sei.cmu.edu/confluence/display/c/POS35-C.+Avoid+race+conditions+while+checking+for+the+existence+of+a+symbolic+link)). You could possibly use `fstat()` instead after `open()`.